iothub/device/devdoc/API Design.md

```diff
@@ -0,0 +1,255 @@
+## Plug and Play convention compatible APIs
+
+### Properties
+
+```csharp
+/// <summary>
+/// Update a single property.
+/// 
+/// Property under the root component:
+/// "reported": {
```

Comment:
If this turns into user facing documentation, I'd rather not document the protocol at this layer so that app can focus on API shape and not implementation details.  ?

Reply:
The examples here are to help us visualize the API and the operation being carried out. I don't expect for this to make it to our official doc comments. We'll probably add a link to the official pnp doc and leave it at that.